processor/transformprocessor/internal/metrics/func_extract_count_metric.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+)
+
+type extractCountMetricArguments struct {
+	Monotonic bool `ottlarg:"0"`
+}
+
+func newExtractCountMetricFactory() ottl.Factory[ottlmetric.TransformContext] {
+	return ottl.NewFactory("extract_count_metric", &extractCountMetricArguments{}, createExtractCountMetricFunction)
+}
+
+func createExtractCountMetricFunction(_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
+	args, ok := oArgs.(*extractCountMetricArguments)
+
+	if !ok {
+		return nil, fmt.Errorf("extractCountMetricFactory args must be of type *extractCountMetricArguments")
+	}
+
+	return extractCountMetric(args.Monotonic)
+}
+
+func extractCountMetric(monotonic bool) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
```

Comment:
How much of this could we easily consolidate into a common function between here and `extractSumMetric`? It seems like we could put most of the sum or count-specific logic into a separate function and have the main section of each OTTL function be common between the two.

Reply:
Probably some of it, but the common part is maybe ~40 LOC, so not that much benefit. The logic is actually a bit different (count is required, while sum isn't and the values are of different types), and having tried it, I think the added abstractions make the code more difficult to understand.

I could separate the logic for determining the aggregation temporality, that's both easy to do and very unlikely to diverge between the two functions.

Reply:
I was curious, so I did a quick test [here](https://github.com/evan-bradley/opentelemetry-collector-contrib/commit/f035b86ef8bd249ad9bc454e15d180988eabc329) and we could share the entirety of both functions without too much overhead.

That said, I agree that for ~40 LOC between only two functions it's not a big deal, so I'll defer to you here. @TylerHelmuth let us know if you feel differently.

Reply:
I am ok with them being separate, it leaves readability a little easier